src/Host/Client/Impl/Host/IRHostBrokerConnector.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Host.Client.Host {
+    public interface IRHostBrokerConnector : IRHostConnector {
+        /// <summary>
+        /// String token that allows to identify broker that is used to connect to RHost
+        /// For local connections, it is a path to the RHost
+        /// For remote connections, it is a URL
+        /// </summary>
+        string BrokerId { get; }
```

Comment:
Maybe actually make it an URL always, and it'll just be `file://` for local? As I recall, you can tease a proper Windows path back out of an URI if it was created with one to begin with.